messages/teleporter/message_handler.go

```diff
@@ -236,6 +240,37 @@ func (m *messageHandler) SendMessage(signedMessage *warp.Message, destinationCli
 		)
 		return err
 	}
+
+	// Wait for the message to be included in a block before returning
+	callCtx, callCtxCancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer callCtxCancel()
+	receipt, err := utils.CallWithRetry[*types.Receipt](
+		callCtx,
+		func() (*types.Receipt, error) {
+			return destinationClient.Client().(ethclient.Client).TransactionReceipt(callCtx, txHash)
+		},
+	)
+	if err != nil {
+		m.logger.Error(
+			"Failed to get transaction receipt",
+			zap.String("destinationBlockchainID", destinationBlockchainID.String()),
+			zap.String("warpMessageID", signedMessage.ID().String()),
+			zap.String("teleporterMessageID", teleporterMessageID.String()),
+			zap.Error(err),
+		)
+		return err
+	}
+	if receipt.Status != types.ReceiptStatusSuccessful {
+		m.logger.Error(
+			"Transaction failed",
+			zap.String("destinationBlockchainID", destinationBlockchainID.String()),
+			zap.String("warpMessageID", signedMessage.ID().String()),
+			zap.String("teleporterMessageID", teleporterMessageID.String()),
+			zap.String("txHash", txHash.String()),
+		)
+		return fmt.Errorf("transaction failed with status: %d", receipt.Status)
+	}
+
 	m.logger.Info(
```

Comment:
Should this message be kept directly after the `SendTx` call? Alternatively, we could re-word it to say "delivered" rather than "sent". Would be great to add the `txHash` to it now that it's available also.

Reply:
Reworded to "delivered". This log message is intended to signal successful delivery.